File: datamodel/src/main/java/datamodel/clinical/PriorMolecularTest.java
package datamodel.clinical;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class PriorMolecularTest {

    @NotNull
    public abstract String test();

    @NotNull
    public abstract String item();

    @Nullable
    public abstract String measure();

    @Nullable
    public abstract String scoreText();

    @Nullable
    public abstract String scoreValuePrefix();

    @Nullable
    public abstract Double scoreValue();

    @Nullable
    public abstract String scoreValueUnit();

    public abstract boolean impliesPotentialIndeterminateStatus();
}

File: algo/src/main/kotlin/com/hartwig/actin/datamodel/clinical/PriorMolecularTest.kt
package com.hartwig.actin.datamodel.clinical

data class PriorMolecularTest(
    val test: String,
    val item: String,
    val measure: String? = null,
    val scoreText: String? = null,
    val scoreValuePrefix: String? = null,
    val scoreValue: Double? = null,
    val scoreValueUnit: String? = null,
    val impliesPotentialIndeterminateStatus: Boolean
)
